### PR TITLE
Transform python-runner into a npm package

### PR DESCRIPTION
**Pull request by jan-warchol**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "build:all": "yarn build:packages && yarn build:refapps && yarn build:docker",
     "build:refapps": "LOCAL_PACKAGES=true lerna run build:refapps",
     "build:packages": "LOCAL_PACKAGES=true lerna run build",
-    "build:all-packages": "LOCAL_PACKAGES=true lerna run build && LOCAL_PACKAGES=true lerna run build:refapps",
+    "build:all-packages": "LOCAL_PACKAGES=true lerna run build && LOCAL_PACKAGES=true lerna run build:refapps && lerna run build-python",
```
Comment (patuwwy) [inline package.json:24]: i suggest to build first, @scramjet/adapters depends on it

File: packages/adapters/src/process-instance-adapter.ts [hunk 2]
```diff
@@ -58,8 +58,8 @@ IComponent {
         if ("python3" in config.engines) {
             this.logger.trace(gotPython);
             const runnerPath = isTSNode
-                ? "../../../python/runner/runner.py"
-                : "../../python/runner/runner.py";
+                ? "../../python-runner/runner.py"
+                : path.resolve(__dirname, require.resolve("@scramjet/python-runner"));
```
Comment (patuwwy) [inline packages/adapters/src/process-instance-adapter.ts:62]: in this case it doesn't matter if it isTSNode or not.
i can't make a suggestion here but i should be just 
```const runnerPath = path.resolve(__dirname, require.resolve("@scramjet/python-runner"));```

File: packages/adapters/src/process-instance-adapter.ts [hunk 3]
```diff
@@ -76,7 +76,9 @@ IComponent {
     getPythonpath() {
         let pythonpath = path.resolve(
             __dirname,
-            isTSNode ? "../../../python_modules" : "../../python_modules"
+            isTSNode
+                ? "../../python-runner/dist"
+                : path.resolve(__dirname, require.resolve("@scramjet/python-runner"), "..")
```
Comment (patuwwy) [inline packages/adapters/src/process-instance-adapter.ts:81]:
```suggestion
           path.resolve(__dirname, require.resolve("@scramjet/python-runner"), "..")
```

File: packages/python-runner/package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "@scramjet/python-runner",
+  "version": "0.17.0",
+  "description": "This package is part of Scramjet Transform Hub. The package executes the remote runners and provides communication with them through abstraction layer provided by adapters.",
```
Comment (patuwwy) [inline packages/python-runner/package.json:4]: Please update description

**Comment by patuwwy** [comment]
Please provide readme

**Review by MichalCz — APPROVED** [review]
Minor proposals.

File: packages/adapters/src/process-instance-adapter.ts (hunk 2 above)
Comment (jan-warchol) [reply packages/adapters/src/process-instance-adapter.ts:62]: Well, I'd like to treat `isTSNode` as "run directly from source" (as opposed to run from `dist/`). At least that's what this code is intended to do - run python runner from source if the whole thing is ran from source.

File: packages/adapters/src/process-instance-adapter.ts (hunk 3 above)
Comment (jan-warchol) [reply packages/adapters/src/process-instance-adapter.ts:81]: similarly to the above, I want to be able to run from where the source code is without building.

File: packages/python-runner/package.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  "main": "./runner.py",
+  "scripts": {
+    "build-python": "mkdir -p dist; cp -r *.py ../../python/scramjet-framework/scramjet/ package.json dist/; yarn prepack",
```
[the remark this answers is not in the document]
Comment (jan-warchol) [reply packages/python-runner/package.json:7]: Changed, please take a look.

File: packages/python-runner/package.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    "postinstall": "pip install --upgrade -r requirements.txt --target dist; cp -r ../../python/scramjet-framework/scramjet/ dist/",
```
[the remark this answers is not in the document]
Comment (jan-warchol) [reply packages/python-runner/package.json:8]: ok

File: package.json (hunk 1 above)
Comment (jan-warchol) [reply package.json:24]: I've renamed the build step to `build`, so it shouldn't be necessary anymore.p

File: packages/python-runner/package.json (hunk 4 above)
Comment (jan-warchol) [reply packages/python-runner/package.json:4]: Done

File: packages/adapters/src/process-instance-adapter.ts (hunk 3 above)
Comment (patuwwy) [reply packages/adapters/src/process-instance-adapter.ts:81]: ok, so why there is `dist` folder involved?

File: packages/python-runner/package.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@scramjet/python-runner",
+  "version": "0.17.0",
+  "description": "This package is part of Scramjet Transform Hub. The package executes user code inside a python process communicating with Transform Hub.",
+  "main": "./runner.py",
+  "scripts": {
+    "preinstall": "pip install --upgrade -r requirements.txt --target dist; cp -r ../../python/scramjet-framework/scramjet/ dist/",
```
Comment (patuwwy) [inline packages/python-runner/package.json:7]: i'm not sure here
if i will do `npm i @scramjet/python-runner` in empty folder it will crash, right?

**Comment by Tatarinho** [comment]
Rebased with release/0.17

File: packages/python-runner/package.json (hunk 7 above)
Comment (MichalCz) [reply packages/python-runner/package.json]: @patuwwy please check again.

**Review by alicja-gruzdz — APPROVED** [review]

**Review by patuwwy — APPROVED** [review]

**Pull request merged**